// Copyright 2024 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


enum IdentitytoolkitBetaTenantMfaConfigStateEnum {
  IdentitytoolkitBetaTenantMfaConfigStateEnumNO_VALUE_DO_NOT_USE = 0;
  IdentitytoolkitBetaTenantMfaConfigStateEnumSTATE_UNSPECIFIED = 1;
  IdentitytoolkitBetaTenantMfaConfigStateEnumDISABLED = 2;
  IdentitytoolkitBetaTenantMfaConfigStateEnumENABLED = 3;
  IdentitytoolkitBetaTenantMfaConfigStateEnumMANDATORY = 4;
}

enum IdentitytoolkitBetaTenantMfaConfigEnabledProvidersEnum {
  IdentitytoolkitBetaTenantMfaConfigEnabledProvidersEnumNO_VALUE_DO_NOT_USE = 0;
  IdentitytoolkitBetaTenantMfaConfigEnabledProvidersEnumPROVIDER_UNSPECIFIED = 1;
  IdentitytoolkitBetaTenantMfaConfigEnabledProvidersEnumPHONE_SMS = 2;
}

message IdentitytoolkitBetaTenant {
  string name = 1;
  string display_name = 2;
  bool allow_password_signup = 3;
  bool enable_email_link_signin = 4;
  bool disable_auth = 5;
  bool enable_anonymous_user = 6;
  IdentitytoolkitBetaTenantMfaConfig mfa_config = 7;
  map<string, string> test_phone_numbers = 8;
  string project = 9;
}

message IdentitytoolkitBetaTenantMfaConfig {
  IdentitytoolkitBetaTenantMfaConfigStateEnum state = 1;
  repeated IdentitytoolkitBetaTenantMfaConfigEnabledProvidersEnum enabled_providers = 2;
}

message ApplyIdentitytoolkitBetaTenantRequest {
  IdentitytoolkitBetaTenant resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteIdentitytoolkitBetaTenantRequest {
  string service_account_file = 1;
  IdentitytoolkitBetaTenant resource = 2;
}

message ListIdentitytoolkitBetaTenantRequest {
  string service_account_file = 1;
  string Project = 2;
}

message ListIdentitytoolkitBetaTenantResponse {
  repeated IdentitytoolkitBetaTenant items = 1;
}

service IdentitytoolkitBetaTenantService {
  rpc ApplyIdentitytoolkitBetaTenant(ApplyIdentitytoolkitBetaTenantRequest) returns (IdentitytoolkitBetaTenant);
  rpc DeleteIdentitytoolkitBetaTenant(DeleteIdentitytoolkitBetaTenantRequest) returns (google.protobuf.Empty);
  rpc ListIdentitytoolkitBetaTenant(ListIdentitytoolkitBetaTenantRequest) returns (ListIdentitytoolkitBetaTenantResponse);
}
